From 9505a2a69aa94aa1d5599e0da91ccd21b9c38e6d Mon Sep 17 00:00:00 2001
From: "Wludzik, Jozef" <jozef.wludzik@intel.com>
Date: Fri, 21 Jun 2024 17:21:58 +0200
Subject: [PATCH 1/3] Fix the compilation issues from gcc-14

Added missing #include <algorithm> to source code reported by gcc-14

(cherry picked from commit ffdc049ca41ad7d8c9555fa95760b7a9f44494a5)
---
 umd/level_zero_driver/core/source/event/event.cpp     | 1 +
 umd/level_zero_driver/ext/source/graph/elf_parser.cpp | 1 +
 umd/level_zero_driver/tools/source/metrics/metric.cpp | 2 ++
 umd/vpu_driver/source/command/vpu_command.cpp         | 1 +
 umd/vpu_driver/source/command/vpu_command_buffer.cpp  | 2 ++
 validation/kmd-test/kmd_test.h                        | 1 +
 6 files changed, 8 insertions(+)

diff --git a/umd/level_zero_driver/core/source/event/event.cpp b/umd/level_zero_driver/core/source/event/event.cpp
index 3cbd08a..193e7ae 100644
--- a/umd/level_zero_driver/core/source/event/event.cpp
+++ b/umd/level_zero_driver/core/source/event/event.cpp
@@ -12,6 +12,7 @@
 #include "vpu_driver/source/utilities/log.hpp"
 #include "vpu_driver/source/utilities/timer.hpp"
 
+#include <algorithm>
 #include <level_zero/ze_api.h>
 #include <thread>
 
diff --git a/umd/level_zero_driver/ext/source/graph/elf_parser.cpp b/umd/level_zero_driver/ext/source/graph/elf_parser.cpp
index 15a797d..f126001 100644
--- a/umd/level_zero_driver/ext/source/graph/elf_parser.cpp
+++ b/umd/level_zero_driver/ext/source/graph/elf_parser.cpp
@@ -12,6 +12,7 @@
 #include "umd_common.hpp"
 #include "vpux_driver_compiler.h"
 
+#include <algorithm>
 #include <cstdint>
 #include <string.h>
 #include <vpux_elf/accessor.hpp>
diff --git a/umd/level_zero_driver/tools/source/metrics/metric.cpp b/umd/level_zero_driver/tools/source/metrics/metric.cpp
index dc48dcb..6d2965d 100644
--- a/umd/level_zero_driver/tools/source/metrics/metric.cpp
+++ b/umd/level_zero_driver/tools/source/metrics/metric.cpp
@@ -8,6 +8,8 @@
 #include "level_zero_driver/tools/source/metrics/metric.hpp"
 #include "vpu_driver/source/utilities/log.hpp"
 
+#include <algorithm>
+
 namespace L0 {
 
 Metric::Metric(zet_metric_properties_t &propertiesInput)
diff --git a/umd/vpu_driver/source/command/vpu_command.cpp b/umd/vpu_driver/source/command/vpu_command.cpp
index f4ca23f..e4772c6 100644
--- a/umd/vpu_driver/source/command/vpu_command.cpp
+++ b/umd/vpu_driver/source/command/vpu_command.cpp
@@ -10,6 +10,7 @@
 #include "vpu_driver/source/device/vpu_device_context.hpp"
 #include "vpu_driver/source/utilities/log.hpp"
 
+#include <algorithm>
 #include <cstddef>
 #include <cstdint>
 #include <vector>
diff --git a/umd/vpu_driver/source/command/vpu_command_buffer.cpp b/umd/vpu_driver/source/command/vpu_command_buffer.cpp
index 4f9b770..1a90915 100644
--- a/umd/vpu_driver/source/command/vpu_command_buffer.cpp
+++ b/umd/vpu_driver/source/command/vpu_command_buffer.cpp
@@ -12,6 +12,8 @@
 #include "vpu_driver/source/command/vpu_copy_command.hpp"
 #include "vpu_driver/source/utilities/log.hpp"
 
+#include <algorithm>
+
 namespace VPU {
 
 VPUCommandBuffer::VPUCommandBuffer(VPUDeviceContext *ctx, VPUBufferObject *buffer, Target target)
diff --git a/validation/kmd-test/kmd_test.h b/validation/kmd-test/kmd_test.h
index ad0baf5..73f4257 100644
--- a/validation/kmd-test/kmd_test.h
+++ b/validation/kmd-test/kmd_test.h
@@ -7,6 +7,7 @@
 
 #pragma once
 
+#include <algorithm>
 #include <libudev.h>
 #include <linux/kernel.h>
 #include <linux/magic.h>
-- 
2.45.2

